src/vehicle.h

```diff
@@ -701,31 +704,54 @@ class vehicle : public JsonSerializer, public JsonDeserializer
     // Loop through engines and generate noise and smoke for each one
     void noise_and_smoke( double load, double time = 6.0 );
 
-    // Calculate area covered by wheels and, optionally count number of wheels
-    float wheels_area (int *cnt = 0) const;
+    /**
+     * Calculates the sum of the area under the wheels of the vehicle.
+     * @param boat If true, calculates the area under "wheels" that allow swimming.
+     */
+    float wheel_area( bool boat ) const;
 
-    // Combined coefficient of aerodynamic and wheel friction resistance of vehicle, 0-1.0.
-    // 1.0 means it's ideal form and have no resistance at all. 0 -- it won't move
-    float k_dynamics () const;
+    /**
+     * Combined coefficient of aerodynamic and wheel friction resistance of vehicle.
+     * @returns A value from 0.0 to 1.0 (exclusive) describing the dynamics coefficient.
+     */
+    float k_dynamics() const;
 
-    // Components of the dynamic coefficient
+    /**
+     * Wheel friction coefficient of the vehicle.
+     * @returns A value from 0.0 to 1.0 (exclusive) describing the friction coefficient.
+     */
     float k_friction () const;
+
+    /**
+     * Air friction coefficient of the vehicle.
+     * @returns A value from 0.0 to 1.0 (exclusive) describing the aerodynamics coefficient.
+     */
     float k_aerodynamics () const;
 
-    // Coefficient of mass, 0-1.0.
-    // 1.0 means mass won't slow vehicle at all, 0 - it won't move
+    /**
+     * Mass coefficient of the vehicle.
+     * Depends on wheel area.
```

Comment:
> The specific results of k_* functions are implementation details

At least `k_aerodynamics`, `k_friction` and `k_mass` are exposed directly to the user via `veh_interact.cpp`. That's the polar opposite of an implementation detail. At an absolute minimum those need to be sufficiently defined so that players can actually make decisions based upon those values.

> As for documenting where are those used: this doesn't happen pretty much anywhere else

The `vehicle` class is in places so poorly documented it's almost impossible to work out what's going on. The options are either to document individual functions or provide a synopsis of how all the parts fit together.